### DISCUSSION TITLE
Question on pairwise negative sampling

### DISCUSSION THREAD

**Discussion by huangtinglin** [pyg-team/pytorch_geometric#6221] [Q&A]
Hi, thank you for your great project!

I am trying to use the negative sampling method for the link prediction task. Can we sample negatives for a batch of nodes, such as a tensor of size 4 where each element is the negative for the corresponding node in the batch?

**Comment by rusty1s** [comment]
You may find the new `neg_sampling="triplet"` strategy useful for this, integrated into the `LinkNeighborLoader` (requries PyG 2.2).

**Reply by huangtinglin** [reply]
That looks good! Thanks.

But I still wonder that is there a function that can generate negatives for a given batch of nodes. For example, neg_nodes = negative_sampling(graph, batch_nodes).

**Reply by rusty1s** [reply]
Yes, there exists a `utils.negative_sampling` function.
